sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -297,110 +287,52 @@ public virtual async Task<Response<UploadLogsResult>> UploadAsync<T>(string rule
                         // Clear any completed blocks from the task list
                         for (int i = 0; i < runningTasks.Count; i++)
                         {
-                            Task<BatchUpload> runningTask = runningTasks[i];
+                            Task<Response> runningTask = runningTasks[i];
                             if (!runningTask.IsCompleted)
                             {
                                 continue;
                             }
-
-                            // Before removing runningTask - check if it has any errors and add to error list
-                            if (runningTask.Result.Error != null)
-                                errors.Add(runningTask.Result.Error);
-
                             // Remove completed task from task list
                             runningTasks.RemoveAt(i);
                             i--;
                         }
                     }
                 }
-
-                // Wait for all the remaining blocks to finish uploading
-                await Task.WhenAll(runningTasks).ConfigureAwait(false);
-
-                // Process all errors after tasks are done to determine status
-                // Will run on a single thread
-                foreach (Task<BatchUpload> task in runningTasks)
+                catch (Exception ex)
```

Comment:
For the async method, we should be able to await all the tasks with Task.WhenAll (and use the linked workaround) to propagate exceptions.